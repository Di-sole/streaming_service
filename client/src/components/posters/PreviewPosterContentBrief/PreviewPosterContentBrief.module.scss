.container {
    font-size: 13px;
    line-height: 16px;
    font-weight: 500;
    color: #d9d7e0;

    &__rating {
        margin-bottom: 12px;
        color: #fff;
        &_integer {
            font-size: 24px;
        }

        &_faction {
            font-size: 20px;
        }
    }
}
