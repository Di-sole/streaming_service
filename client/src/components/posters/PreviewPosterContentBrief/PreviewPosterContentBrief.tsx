import IMovie from '@/models/IMovie';
import styles from './PreviewPosterContentBrief.module.scss';
import { declineWord, firstCapitalLetter } from '@/utils/functions';
import { useRouter } from 'next/router';
import { useTranslation } from 'next-i18next';

interface PreviewPosterContentBriefProps {
    movie: IMovie;
    className?: string;
}

const PreviewPosterContentBrief = ({ movie, className }: PreviewPosterContentBriefProps) => {
    const { locale } = useRouter();
    const { t } = useTranslation('moviesPage');
    return (
        <div className={[styles.container, className].join(' ')}>
            {movie.ratingKp && (
                <div className={styles.container__rating}>
                    <span className={[styles.container__rating, styles.container__rating_integer].join(' ')}>
                        {`${movie.ratingKp.toFixed(1)}`.slice(0, 1)}
                    </span>
                    <span className={[styles.container__rating, styles.container__rating_fraction].join(' ')}>
                        {`${movie.ratingKp.toFixed(1)}`.slice(1, 3)}
                    </span>
                </div>
            )}
            <div className={styles.container__text}>
                <p>
                    {movie.year},{' '}
                    {locale === 'en' && movie.countries[0].enName ? movie.countries[0].enName : movie.countries[0].name}
                    , {firstCapitalLetter(locale === 'ru' ? movie.genres[0].name : movie.genres[0].enName)}
                </p>
                <p>{`${movie.movieLength} ${declineWord(movie.movieLength, [
                    t('minutes.0'),
                    t('minutes.1'),
                    t('minutes.2'),
                ])}`}</p>
            </div>
        </div>
    );
};

export default PreviewPosterContentBrief;
