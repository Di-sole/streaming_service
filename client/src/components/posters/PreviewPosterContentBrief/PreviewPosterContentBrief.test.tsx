import { render, screen } from '@testing-library/react';
import PreviewPosterContentBrief from './PreviewPosterContentBrief';
import { movieTest } from '../../../testAsserts/testItems';

jest.mock('next-i18next', () => ({
    useTranslation: () => {
        return {
            t: (str: string) => str,
            i18n: {
                changeLanguage: () => new Promise(() => {}),
            },
        };
    },
}));

jest.mock('next/router', () => ({
    useRouter() {
        return {
            pathname: '',
        };
    },
}));

describe('PREVIEWPOSTERCONTENTBRIEF SNAPSHOTS TESTS', () => {
    test('PreviewPosterContentBrief should not be changed', () => {
        render(<PreviewPosterContentBrief movie={movieTest} />);

        const id = screen.getByTestId('previewPosterContentBrief');
        expect(id).toMatchSnapshot();
    });
});
