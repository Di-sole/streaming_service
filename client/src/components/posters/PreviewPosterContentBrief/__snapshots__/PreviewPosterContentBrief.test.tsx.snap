// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PREVIEWPOSTERCONTENTBRIEF SNAPSHOTS TESTS PreviewPosterContentBrief should not be changed 1`] = `
<div
  class="container "
  data-testid="previewPosterContentBrief"
>
  <div
    class="container__rating"
  >
    <span
      class="container__rating container__rating_integer"
    >
      8
    </span>
    <span
      class="container__rating container__rating_fraction"
    >
      .8
    </span>
  </div>
  <div
    class="container__text"
  >
    <p>
      2011
      , 
      France
      ,
       
      Drama
    </p>
    <p>
      112 minutes.2
    </p>
  </div>
</div>
`;
